backend/test/users.test.js

```diff
@@ -1,219 +1,288 @@
+                .send(loginInfo)
+
+            // THEN a response with status code 401 should be returned along with an error message
+            assert.equal(response.statusCode, 401)
+            assert.equal(response.body.error, 'Incorrect email or password')
+            done()
+        })
+    })
+
+    describe('Logging in with wrong email', () => {
+        it('Should return a response status code of 401 and error message', async (done) => {
+            // GIVEN a created user
+            const password = 'PASSWORD'
+            const randomUsername = (Math.random() + 1).toString(36).substring(7)
+            const email = `test@${randomUsername}.com`
+
+            await models.users.create({
+                username: randomUsername,
+                email: email,
+                password: password,
+            })
+
+            // WHEN the user attempts to log in with the wrong email
+            const loginInfo = {
+                email: 'wrong@email.com',
+                password: password,
+            }
+            const response = await request('http://localhost:8000/api')
+                .post('/users/authenticate')
+                .send(loginInfo)
+
+            // THEN a response with status code 401 should be returned along with an error message
+            assert.equal(response.statusCode, 401)
+            assert.equal(response.body.error, 'Incorrect email or password')
+            done()
+        })
+    })
+
+    describe('Testing getUsersById endpoint', () => {
+        it('Should return a 200 status response', async (done) => {
+            // GIVEN a created user
+            const password = 'PASSWORD'
+            const randomUsername = (Math.random() + 1).toString(36).substring(7)
+            const email = `test@${randomUsername}.com`
+            const profilePic = 'https://imgur.com/gallery/zIMAzsV'
+            const profileBanner = 'https://imgur.com/gallery/RstwImS'
+            const bio = 'Bio'
+            const followers = [1, 2]
+            const following = [3, 4]
+
+            const result = await models.users.create({
+                username: randomUsername,
+                email: email,
+                password: password,
+                profilePic,
+                profileBanner,
+                bio,
+                followers,
+                following,
+            })
+
+            const loginInfo = {
+                email: email,
+                password: password,
+            }
+
+            const auth = await request('http://localhost:8000/api')
+                .post('/users/authenticate')
+                .send(loginInfo)
+
+            // user viewing itself
+            const response = await request('http://localhost:8000/api')
+                .get(`/users/${result.id}`)
+                .set('Authorization', `Bearer ${auth.body.authToken}`)
+
+            assert.equal(response.statusCode, 200)
+            assert.equal(response.body.followers, 2)
```

Comment:
Could we compare the whole DTO object and see if it's the one we expect for this test and the test below?